clients/wallet/src/stellar_wallet.rs

```diff
@@ -387,10 +408,65 @@ mod test {
 		"SCV7RZN5XYYMMVSWYCR4XUMB76FFMKKKNHP63UTZQKVM4STWSCIRLWFJ";
 	const IS_PUBLIC_NETWORK: bool = false;
 
+	const DEFAULT_STROOP_FEE_PER_OPERATION: u32 = 100;
+
+	impl StellarWallet {
+		async fn is_account_exist(&self) -> bool {
+			self.client
+				.get_account(self.get_public_key(), self.is_public_network)
+				.await
+				.is_ok()
+		}
+		
+		/// merges the wallet's account with the specified destination.
+		/// Exercise prudence when using this method, as it automatically removes the source account
+		/// once operation is successful.
+		async fn merge_account(
+			&mut self,
+			destination_address: PublicKey,
+		) -> Result<TransactionResponse, Error> {
+			let account_merge_op =
+				account_merge_operation(destination_address, self.secret_key.get_public().clone())?;
+
+			self.send_to_address([9u8; 32], DEFAULT_STROOP_FEE_PER_OPERATION, vec![account_merge_op])
+				.await
+		}
+
+		fn create_payment_envelope(
+			&self,
+			destination_address: PublicKey,
+			asset: Asset,
+			stroop_amount: StellarStroops,
+			request_id: [u8; 32],
+			stroop_fee_per_operation: u32,
+			next_sequence_number: SequenceNumber,
+		) -> Result<TransactionEnvelope, Error> {
+			let public_key = self.get_public_key();
+			// create payment operation
+			let payment_op = create_payment_operation(
+				destination_address,
+				asset,
+				stroop_amount,
+				public_key.clone(),
+			)?;
+
+			self.create_envelope(
+				request_id,
+				stroop_fee_per_operation,
+				next_sequence_number,
+				vec![payment_op],
+			)
+		}
+	}
+
 	fn wallet(storage: &str) -> Arc<RwLock<StellarWallet>> {
```

Comment:
Nit: I'd rename to `wallet_with_secret_key_for_storage(storage: &str)` or similar. Seeing just `wallet("")` being called in the tests is confusing.

Reply:
renamed